### PR TITLE
deprecating the need for interp_u method in integrators

### PR DESCRIPTION
**Pull request by drgona**
Propose simplification of ODE integrators in integrators.py

### CODE REVIEW

**Comment by drgona** [comment]
Before merge all non-autonomous ODE system ID and control examples needs to be updated with removed interp_u

**Comment by madelynshapiro** [comment]
@drgona I recommend marking this PR as a !Draft to block merging until all remaining tasks are complete

File: src/neuromancer/dynamics/integrators.py [hunk 1]
```diff
@@ -16,31 +16,27 @@ def __init__(self, block, interp_u=None, h=1.0):
...
         self.in_features, self.out_features = block.in_features, block.out_features
-        self.interp_u = interp_u  # offline interpolant
         self.h = h
-        if self.interp_u is not None:
-            self.state = lambda x, tq, t, u: torch.cat([x, self.interp_u(tq, t, u)], dim=-1)
-        else:
-            self.state = lambda x, tq, t, u: x
+
+        if interp_u is not None:
+            warnings.warn('interp_u method is deprecated, it has no effect, and will be removed in the next release.')
```
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:28]:
```suggestion
            warnings.warn('interp_u argument is deprecated, it has no effect, and will be removed in the next release.', FutureWarning)
```

https://docs.python.org/3/library/exceptions.html#FutureWarning

File: src/neuromancer/dynamics/integrators.py [hunk 2]
```diff
@@ -80,9 +76,10 @@ def __init__(self, block, interp_u=None, h=0.001, method='euler'):
         self.method = method
         self.adjoint_params = torchdiffeq._impl.adjoint.find_parameters(self.block)
 
-    def integrate(self, x, u, t):
-        timepoints = torch.tensor([t[0], t[0] + self.h])
-        rhs_fun = lambda t, x: self.block(self.state(x, t, t, u))
+    def integrate(self, x, *args):
+        t_0 = 0.0
```
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:80]: Is there any case where the user might want to set a non-zero `t_0`?

File: src/neuromancer/dynamics/integrators.py [hunk 3]
```diff
@@ -162,21 +144,16 @@ def __init__(self, block, interp_u=None, h=1.0):
...
 
-    def integrate(self, x, u, t):
+    def integrate(self, x, *args):
         h = self.h
-        k1 = self.block(self.state(x, t, t, u))                    # k1 = f(x_i, t_i)
-        k2 = self.block(self.state(x + h*k1/2.0, t + h/2, t, u))   # k2 = f(x_i + 0.5*h*k1, t_i + 0.5*h)
-        k3 = self.block(self.state(x + h*k2/2.0, t + h/2, t, u))   # k3 = f(x_i + 0.5*h*k2, t_i + 0.5*h)
-        k4 = self.block(self.state(x + h*k3, t + h, t, u))         # k4 = f(y_i + h*k3, t_i + h)
+        k1 = self.block(x, *args)                    # k1 = f(x_i, t_i)
+        k2 = self.block(x + h*k1/2.0, *args)         # k2 = f(x_i + 0.5*h*k1, t_i + 0.5*h)
```
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:154]: The old code modifies `t` to pass the `tq` arg to `self.state` for `k2` - `k4`. Does this need to be handled in the new implementation or is `tq` functionality being deprecated?

Comment (madelynshapiro) [reply src/neuromancer/dynamics/integrators.py:154]: If deprecating, update function def inline comments to remove `t_i` term

Comment (madelynshapiro) [reply src/neuromancer/dynamics/integrators.py:154]: Same comment applies to all other classes' `integrate` methods implemented below

**Comment by madelynshapiro** [comment]
@drgona Almost ready to merge:

- [x] Tests passing
- [ ] Resolve minor review comments above
- [ ] Update version number to 1.4.1 in `pyproject.toml`
- [ ] Summarize changes for 1.4.1 in main `README.md`

**Review by madelynshapiro — APPROVED** [review]

**Pull request merged**